src/xpython_interpreter.cpp

```diff
@@ -159,7 +136,39 @@ namespace xpyt
         return result;
     }
 
-    void python_interpreter::input_reply_impl(const std::string& /*value*/)
+    void xpython_interpreter::input_reply_impl(const std::string& /*value*/)
+    {
+    }
+
+    void xpython_interpreter::redirect_output()
     {
+        // In Python:
+        // import sys and import xeus_python_logger
+        py::module sys = py::module::import("sys");
+        py::module xeus_python_logger = py::module::import("xeus_python_logger");
+
+        // Create XPythonLogger instance
+        py::object out_logger = xeus_python_logger.attr("XPythonLogger")();
+        py::object err_logger = xeus_python_logger.attr("XPythonLogger")();
+
+        py::cpp_function publish_stdout_stream = [this](const std::string& message){
+            this->publish_stream("stdout", message);
+        };
+
+        py::cpp_function publish_stderr_stream = [this](const std::string& message){
+            this->publish_stream("stderr", message);
+        };
+
+        // Add publish_stream as a logger function
+        out_logger.attr("add_logger")(publish_stdout_stream);
+        err_logger.attr("add_logger")(publish_stderr_stream);
+
+        // Save current stdout/stderr
+        sys.attr("__stdout__") = sys.attr("stdout");
+        sys.attr("__stderr__") = sys.attr("stderr");
```

Comment:
@Carreau are you sure this is needed? From the [documentation](https://docs.python.org/2/library/sys.html#sys.__stderr__):

> These objects contain the original values of stdin, stderr and stdout at the start of the program. They are used during finalization, and could be useful to print to the actual standard stream no matter if the sys.std* object has been redirected.

It seems to me that `__stdout__` and `__stderr__` already contains the `stdout` and `stderr`